packages/core/src/connectors/facebook/index.ts

```diff
@@ -0,0 +1,130 @@
+/**
+ * Reference: Manually Build a Login Flow
+ * https://developers.facebook.com/docs/facebook-login/manually-build-a-login-flow
+ */
+import got, { RequestError as GotRequestError } from 'got';
+import { stringify } from 'query-string';
+import { z } from 'zod';
+
+import {
+  accessTokenEndpoint,
+  authorizationEndpoint,
+  scope,
+  userInfoEndpoint,
+} from '@/connectors/facebook/constant';
+import {
+  ConnectorError,
+  ConnectorErrorCodes,
+  ConnectorMetadata,
+  ConnectorType,
+  GetAccessToken,
+  GetAuthorizationUri,
+  GetUserInfo,
+  ValidateConfig,
+} from '@/connectors/types';
+import { getConnectorConfig, getConnectorRequestTimeout } from '@/connectors/utilities';
+import assertThat from '@/utils/assert-that';
+
+export const metadata: ConnectorMetadata = {
+  id: 'facebook',
+  type: ConnectorType.Social,
+  name: {
+    en: 'Sign In with Facebook',
+    zh_CN: 'Facebook 登录',
+  },
+  // TODO: add the real logo URL (LOG-1823)
+  logo: './logo.png',
+  description: {
+    en: 'Sign In with Facebook',
+    zh_CN: 'Facebook 登录',
+  },
+};
+
+const facebookConfigGuard = z.object({
+  clientId: z.string(),
+  clientSecret: z.string(),
+});
+
+type FacebookConfig = z.infer<typeof facebookConfigGuard>;
+
+export const validateConfig: ValidateConfig = async (config: unknown) => {
+  const result = facebookConfigGuard.safeParse(config);
+
+  if (!result.success) {
+    throw new ConnectorError(ConnectorErrorCodes.InvalidConfig, result.error.message);
+  }
+};
+
+export const getAuthorizationUri: GetAuthorizationUri = async (redirectUri, state) => {
+  const config = await getConnectorConfig<FacebookConfig>(metadata.id);
+
+  return `${authorizationEndpoint}?${stringify({
+    client_id: config.clientId,
+    redirect_uri: redirectUri,
+    response_type: 'code',
+    state,
+    scope, // Only support fixed scope for v1.
+  })}`;
+};
+
+export const getAccessToken: GetAccessToken = async (code, redirectUri) => {
+  type AccessTokenResponse = {
+    access_token: string;
+    token_type: string;
+    expires_in: number;
+  };
+
+  const { clientId: client_id, clientSecret: client_secret } =
+    await getConnectorConfig<FacebookConfig>(metadata.id);
+
+  const { access_token: accessToken } = await got
+    .get(accessTokenEndpoint, {
+      searchParams: {
+        code,
+        client_id,
+        client_secret,
+        redirect_uri: redirectUri,
+      },
+      timeout: await getConnectorRequestTimeout(),
+    })
+    .json<AccessTokenResponse>();
+
+  assertThat(accessToken, new ConnectorError(ConnectorErrorCodes.SocialAuthCodeInvalid));
+
+  return accessToken;
+};
+
+export const getUserInfo: GetUserInfo = async (accessToken: string) => {
+  type UserInfoResponse = {
+    id: string;
+    email?: string;
+    name?: string;
+    picture?: { data: { url: string } };
```

Comment:
by reading the doc, found that a boolean property `is_silhouette` is with the same level as `url` does. Wondering what does it mean, or will it affect the validity of `url`?

Reply:
`is_silhouettebool`: True if the profile picture is the default 'silhouette' picture.

We can ignore this field for now.

https://developers.facebook.com/docs/graph-api/reference/profile-picture-source/#fields